File: src/main/java/com/backend/app/userservice/models/SignUpRequest.java
package com.backend.app.userservice.models;

import com.fasterxml.jackson.annotation.JsonProperty;

import lombok.Data;

@Data
public class SignUpRequest {
     @JsonProperty("username")
    private String username;

    @JsonProperty("firstname")
    private String firstname;

    @JsonProperty("lastname")
    private String lastname;

    @JsonProperty("password")
    private String password;

    @JsonProperty("email")
    private String email;
}

File: user-svc/UserService/Apps/Users/Models/Requests/SignUpRequest.cs
using System.Text.Json.Serialization;

namespace UserService.Apps.Users.Models.Requests
{
    /* 
     {
        "username": "firstuser",
        "firstname": "First",
        "lastname": "lastName",
        "password": "pass@word1",
        "email": "example@gmail.com",
        "phoneOne": "011459844874"
    }
     */
    public class SignUpRequest
    {
        [JsonPropertyName("username")]
        public required string Username { get; set; }
    
        [JsonPropertyName("firstname")]
        public required string Firstname { get; set; }

        [JsonPropertyName("lastname")]
        public string? Lastname { get; set; }

        [JsonPropertyName("email")]
        public required string Email { get; set; }

        [JsonPropertyName("password")]
        public required string Password { get; set; }

        [JsonPropertyName("phoneOne")]
        public string? PhoneOne { get; set; }
    }
}
